core/src/main/java/org/mule/runtime/core/util/collection/EventToMessageSequenceSplittingStrategy.java

```diff
@@ -39,35 +46,56 @@ public EventToMessageSequenceSplittingStrategy(ExpressionSplittingStrategy expre
   @Override
   @SuppressWarnings({"unchecked", "rawtypes"})
   public MessageSequence<?> split(Event event) {
-    Message msg = event.getMessage();
-    Object payload = msg.getPayload().getValue();
-    if (payload instanceof MessageSequence<?>) {
-      return ((MessageSequence<?>) payload);
-    }
-    if (payload instanceof Iterator<?>) {
-      return new IteratorMessageSequence(((Iterator<Object>) payload));
+    if (expressionSplitterStrategy.hasDefaultExpression()) {
```

Comment:
Right, but how does studio know if the result of split is `Entry` or `Object` etc?  That was one of the issues with using `Entry` before, there was no way for user to know what type was, and they'd typically expect value...